.github/workflows/doxygen.yaml

```diff
@@ -0,0 +1,46 @@
+name: Doxygen check
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - '*'
+
+env:
+  BUILD_DEPENDENCIES: cmake build-essential checkinstall zlib1g-dev libssl-dev libfmt-dev
+  BUILD_DIRECTORY: build
+
+jobs:
+  doxygen-check:
+    name: Doxygen Check
+    runs-on: ubuntu-22.04
+    timeout-minutes: 30
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          submodules: recursive
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update && \
+          sudo apt-get install -y ${{ env.BUILD_DEPENDENCIES }} clang-14 doxygen graphviz
+
+      - name: Check doxygen
+        env:
+          CC: clang
+          CXX: clang++
+        run: |
+          cmake -S . -B ${{ env.BUILD_DIRECTORY }} -DCMAKE_BUILD_TYPE=Debug -DDOXYGEN_BUILD_ENABLED=ON
+          cmake --build ${{ env.BUILD_DIRECTORY }} --target doxygen -- --no-print-directory
+          doxygen ${{ env.BUILD_DIRECTORY }}/doxygen/Doxyfile
+
+      - name: Upload Doxygen Artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: doxygen-docs
+          path: ${{ env.BUILD_DIRECTORY }}/doxygen/docs/html
+          if-no-files-found: error
```

Comment:
Investigate to upload in a seperate job, which is triggered manually only, but the check is trigger all the time